### PR TITLE
Update docs and flake

### PR DESCRIPTION
**Pull request by deemp**
Previously: part of #288

- Update docs
  - Refactor files structure
  - Add TOC
- Update flake
  - Update nixpkgs
  - Use flake-utils
  - Refactor to use flake-utils

### CODE REVIEW

File: docs/book.toml [hunk 1]
```diff
@@ -2,10 +2,12 @@
 authors = ["zimbatm"]
 language = "en"
 multilingual = false
-src = "."
+src = "src"
```
Comment (zimbatm) [inline docs/book.toml:5]: What is the motivation for moving this one level down? It's one more level to traverse when navigating the code.

File: flake.nix [hunk 2]
```diff
@@ -1,48 +1,76 @@
...
-      templates = rec {
-        toml = {
-          path = ./templates/toml;
-          description = "nix flake new my-project -t github:numtide/devshell";
-        };
-        flake-parts = {
-          path = ./templates/flake-parts;
-          description = "nix flake new my-project -t github:numtide/devshell#flake-parts";
+  # nix flake update --recreate-lock-file
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/efc960b6d6a6498c23868f4ba59fcb8bb51c9861";
```
Comment (zimbatm) [inline flake.nix:5]: Can you point to a channel instead, to make it easier to bump nixpkgs?

**Review by zimbatm — COMMENTED** [review]
almost ready

File: docs/book.toml (hunk 1 above)
Comment (deemp) [reply docs/book.toml:5]: This structure is described in the [mdbook docs](https://rust-lang.github.io/mdBook/guide/creating.html#source-files)

File: flake.nix (hunk 2 above)
Comment (deemp) [reply flake.nix]: Done in https://github.com/numtide/devshell/pull/289/commits/bd708b4c66bd1d00459136340c8f90330b0acb6e

**Comment by zimbatm** [comment]
thanks!

**Comment by sigprof** [comment]
Before this change it was possible to use the devshell flake without overlays (#162, #165):
```nix
    devshell.inputs.nixpkgs.follows = "nixpkgs";
```
```nix
          devShells.default = inputs.devshell.legacyPackages.${system}.mkShell {
            # ... configuration ...
          };
```

But this PR removed `legacyPackages`.  What would be the proper way to use the devshell flake without overlays now?

The following works, but looks like a hack due to the non-flake import (and the `inputs` and `nixpkgs` arguments for `default.nix` don't seem to be documented):
```nix
          devshell = import inputs.devshell {
            inherit system;
            inputs = null;
            nixpkgs = inputs.nixpkgs.legacyPackages.${system};
          };
```
```nix
          devShells.default = devshell.mkShell {
            # ... configuration ...
          };
```

Is migrating to flake-parts the only supported way forward?

**Comment by deemp** [comment]
Thank you! It's a regression. The only way is to restore `legacyPackages`.

File: docs/default.nix [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
-      (source.matchExt "md")
-    ];
-  };
+  src =
+    let fs = lib.fileset; in
+    fs.toSource {
+      root = ./.;
+      fileset = fs.unions [
+        (fs.fileFilter (file: file.hasExt "md") ./src)
+        (fs.fileFilter (file: true) ./theme)
```
Comment (infinisil) [inline docs/default.nix:16]: Oh just randomly seeing this: There's no need to use `fileFilter` when you want to include every file of a directory, `./theme` by itself is also coerced to a file set to contain all its files :)

```suggestion
        ./theme
```

**Pull request merged**